src/probnum/filtsmooth/statespace/discrete_transition.py

```diff
@@ -31,80 +34,30 @@ class DiscreteGaussian(trans.Transition):
     :class:`DiscreteGaussianLinearModel`
     """
 
-    def __init__(self, dynamicsfun, diffmatfun, jacobfun=None):
-        self._dynamicsfun = dynamicsfun
-        self._diffmatfun = diffmatfun
-        self._jacobfun = jacobfun
+    def __init__(
+        self,
+        dynamicsfun: Callable[[FloatArgType, np.ndarray], np.ndarray],
+        diffmatfun: Callable[[FloatArgType], np.ndarray],
+        jacobfun: Optional[Callable[[FloatArgType, np.ndarray], np.ndarray]] = None,
+    ):
+        self.dynamicsfun = dynamicsfun
+        self.diffmatfun = diffmatfun
 
-    def transition_realization(self, real, start, stop=None, **kwargs):
-        newmean = self._dynamicsfun(start, real)
-        newcov = self._diffmatfun(start)
+        def if_no_jacobian(t, x):
+            raise NotImplementedError
+
+        self.jacobfun = jacobfun if jacobfun is not None else if_no_jacobian
+
+    def transition_realization(self, real, start, **kwargs):
+        # **kwargs swallow all irrelevant arguments for this function.
```

Comment:
now, this is the only purpose of kwargs. Do you think this is cleaner, too? Shall I remove the comment before merging?

Reply:
Yes I think this is cleaner than before :) It's also fine for me to remove the comments before merging.